keras_cv/models/backbones/resnet_v1/resnet_v1_backbone.py

```diff
@@ -0,0 +1,540 @@
+    )(x)
+    x = layers.BatchNormalization(
+        axis=BN_AXIS, epsilon=BN_EPSILON, name=name + "_1_bn"
+    )(x)
+    x = layers.Activation("relu", name=name + "_1_relu")(x)
+
+    x = layers.Conv2D(
+        filters,
+        kernel_size,
+        padding="SAME",
+        use_bias=False,
+        name=name + "_2_conv",
+    )(x)
+    x = layers.BatchNormalization(
+        axis=BN_AXIS, epsilon=BN_EPSILON, name=name + "_2_bn"
+    )(x)
+    x = layers.Activation("relu", name=name + "_2_relu")(x)
+
+    x = layers.Conv2D(4 * filters, 1, use_bias=False, name=name + "_3_conv")(x)
+    x = layers.BatchNormalization(
+        axis=BN_AXIS, epsilon=BN_EPSILON, name=name + "_3_bn"
+    )(x)
+
+    x = layers.Add(name=name + "_add")([shortcut, x])
+    x = layers.Activation("relu", name=name + "_out")(x)
+    return x
+
+
+def apply_stack(
+    x,
+    filters,
+    blocks,
+    stride=2,
+    name=None,
+    block_type="block",
+    first_shortcut=True,
+):
+    """A set of stacked residual blocks.
+
+    Args:
+      x: input tensor.
+      filters: int, filters of the layer in a block.
+      blocks: int, blocks in the stacked blocks.
+      stride: int, stride of the first layer in the first block. Defaults to 2.
+      block_type: string, one of "basic_block" or "block". The block type to
+            stack. Use "basic_block" for ResNet18 and ResNet34.
+      first_shortcut: bool. Use convolution shortcut if `True` (default),
+            otherwise uses identity or pooling shortcut, based on stride.
+
+    Returns:
+      Output tensor for the stacked blocks.
+    """
+
+    if name is None:
+        name = "v1_stack"
+
+    if block_type == "basic_block":
+        block_fn = apply_basic_block
+    elif block_type == "block":
+        block_fn = apply_block
+    else:
+        raise ValueError(
+            """`block_type` must be either "basic_block" or "block". """
+            f"Received block_type={block_type}."
+        )
+
+    x = block_fn(
+        x,
+        filters,
+        stride=stride,
+        name=name + "_block1",
+        conv_shortcut=first_shortcut,
+    )
+    for i in range(2, blocks + 1):
+        x = block_fn(
+            x, filters, conv_shortcut=False, name=name + "_block" + str(i)
+        )
+    return x
+
+
+@keras.utils.register_keras_serializable(package="keras_cv.models")
+class ResNetBackbone(Backbone):
+    """Instantiates the ResNet architecture.
+
+    Reference:
+        - [Deep Residual Learning for Image Recognition](https://arxiv.org/abs/1512.03385)
+
+    The difference in ResNetV1 and ResNetV2 rests in the structure of their
+    individual building blocks. In ResNetV2, the batch normalization and
+    ReLU activation precede the convolution layers, as opposed to ResNetV1 where
+    the batch normalization and ReLU activation are applied after the
+    convolution layers.
+
+    For transfer learning use cases, make sure to read the [guide to transfer
```

Comment:
Will this render? My understanding is that the link needs to be on one line

Reply:
Done.